src/Symfony/Component/ExpressionLanguage/TokenStream.php

```diff
@@ -56,6 +56,10 @@ public function next()
 
         ++$this->position;
 
+        if (!isset($this->tokens[$this->position])) {
+            throw new SyntaxError('Unexpected end of expression', $this->current->cursor, $this->expression);
```

Comment:
Shouldn't we move the safe guard that was already there? I don't see how the first exception could ever be thrown. It looks to me like this was just mistakenly placed before the pointer is advanced.

Reply:
done